resolwe/flow/models/data.py

```diff
@@ -457,6 +487,59 @@ def delete(self, *args, **kwargs):
 
         Storage.objects.filter(pk__in=storage_ids, data=None).delete()
 
+    def is_duplicate(self):
+        """Return True if data object is a duplicate."""
+        return bool(self.duplicated)
+
+    def duplicate(self, contributor=None):
+        """Duplicate (make a copy)."""
+        if self.status not in [self.STATUS_DONE, self.STATUS_ERROR]:
+            raise ValidationError('Data object must have done or error status to be duplicated')
+
+        duplicate = Data.objects.get(id=self.id)
+        duplicate.pk = None
+        duplicate.slug = None
+        name_max_length = Data._meta.get_field('name').max_length
+        duplicate.name = 'Copy of {}'.format(self.name)[:name_max_length]
+        duplicate.duplicated = now()
+        if contributor:
+            duplicate.contributor = contributor
+
+        duplicate._perform_save(force_insert=True)  # pylint: disable=protected-access
+
+        # TODO
+        # Create duplication data migration history entry.
+
+        assign_contributor_permissions(duplicate)
+
+        # Override fields that are automatically set on create.
+        duplicate.created = self.created
+        duplicate._perform_save()  # pylint: disable=protected-access
+
+        if self.location:
+            self.location.data.add(duplicate)  # pylint: disable=no-member
+
+        duplicate.storages.set(self.storages.all())  # pylint: disable=no-member
+
+        migration_queryset = DataMigrationHistory.objects.filter(data=self).order_by('created')
```

Comment:
Change the query from
`DataMigrationHistory.objects.filter(data=self).order_by('created')`
to
`self.migration_history.all()`?